### ISSUE TITLE
no-unused-vars false negative for `type name = 'string' | null`

### ISSUE THREAD

**Issue by marcusjwhelan** [typescript-eslint/typescript-eslint#4588]
- [ x] I have tried restarting my IDE and the issue persists.
- [x ] I have updated to the latest version of the packages.
- [x ] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    '@typescript-eslint/no-unused-vars': 1,
  }
}
```

```TS
type thisType = 'hello' | 'world' | null
const handle = (input: thisType) => {}
```

TsConfig, There is a error if I set `noUnusedParameters` to true, however I want a warning not an error.
```json
{
  "compilerOptions": {
    // Target latest version of ECMAScript.
    "target": "es2020",
    // path to output directory
    "outDir": "./public/",
    // enable strict null checks as a best practice
    "strictNullChecks": true,
    // Search under node_modules for non-relative imports.
    "moduleResolution": "node",
    // Process & infer types from .js files.
    "allowJs": true,
    // Don't emit; allow Babel to transform files.
    "noEmit": true,
    // Enable strictest settings like strictNullChecks & noImplicitAny.
    "strict": true,
    // Import non-ES modules as default imports.
    "esModuleInterop": true,
    // use typescript to transpile jsx to js
    "jsx": "react",
    "baseUrl": "./src",
    "lib": [
      "es2020",
      "dom.iterable",
      "es2016.array.include",
      "es2019.object",
      "dom"
    ],
    "module": "esnext",
    "downlevelIteration": false,
    "checkJs": false,
    "removeComments": true,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
  }
}
```

**Expected Result**

a warning underline.

**Actual Result**

no underline of any kind. Only hovering over it will show that it is unused. A normal type of 
string, number, or anything besides a given type will have the correct underline warning.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.12.1` |
| `@typescript-eslint/parser`        | `5.12.1` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `16.14.0` |

ESLint 8 breaks a lot of rules since the rules haven't been updated I have to stay on 7. You can see this on stackoverflow

https://stackoverflow.com/questions/69578685/eslint-broken-rules-with-suggestions-must-set-the-meta-hassuggestions-propert

**Comment by bradzacher** [comment]
Unless I'm misunderstanding you... This isn't a false negative.
This is correct - `thisType` is ***used*** in the function.

Could you please provide more information?

**Comment by bradzacher** [comment]
your original example only had one argument... you should make sure you include all relevant information when filing an issue.

This is working as expected.
Please read the rule's documentation https://eslint.org/docs/rules/no-unused-vars#options

You will find that `args: 'after-used'` is the default config.
